system/system.go

```diff
@@ -80,11 +82,27 @@ func (c *Cmd) StdoutPipe() (io.ReadCloser, error) {
 	return c.Cmd.StdoutPipe()
 }
 
+type cmdLogger struct {
+	cmdName string
+	stream  string
+}
+
+func (c *cmdLogger) Write(buf []byte) (int, error) {
+	lines := strings.Split(string(buf), "\n")
+	for _, line := range lines {
+		if len(line) > 0 {
+			log.Infof("Output (%s) from command %q: %s", c.stream, c.cmdName, line)
```

Comment:
Actually, I was envisioning something more along the lines of:
```
Collected output from command...:
line
line
.
.
.
line
```
Like, this is really one log.

What do you think?

Reply:
We should not collect all the output from the command, because this can grow without bound, and we lose both timestamps and ordering. We could perhaps do it as you say for each block that is written, but then it will be:

```
Collected output from command...:
line
line
.
.
.
line
Some other Mender log output...
Collected output from command...:
line
line
.
.
.
line
```

because `Write()` will be called multiple times. So maybe a bit inconsistent. Do you think it's better? Is so, I should at least check it does not cause any problems in the systemd or deployment log.

Reply:
Good point, I guess we should first decide

1. if you also think this is better (the collected output).

-> If so, then we can make the writer buffered without too much trouble I guess :) ?

Reply:
>     1. if you also think this is better (the collected output).

Copy pasting is more annoying with the current method, but I still think I lean towards that one. Strictly speaking it is what I would expect from a logger. It's ugly, but it's consistent, and every log line will look the same! :-)

Oh, and I just discovered that the Update Module output is already printed this way.

> If so, then we can make the writer buffered without too much trouble I guess :) ?

It's already buffered before `Write()` is called. We definitely should not add additional buffering, because then we lose timestamps and ordering as I mentioned earlier, even if the buffer is small. For example it might print two things several seconds apart, and only on the second print is the buffer full and we print it. This will be misleading in the log.

Reply:
Makes sense. I yield :)